File: C#/SEP3/Shared/domain/Product.cs
namespace Shared;

public class Product
{
    public long id { get; set; }
    public string name { get; set; } 
    public string description { get; set; }
    public double price { get; set; }
    public int stock { get; set; }
    public string image { get; set; }
    public string? ingredients  { get; set; }
    
    public Category category { get; set; }
   public Product(long id, string name, string description, double price, int stock, string image, string ingredients, Category category)
   {
       this.id = id;
       this.name = name;
       this.description = description;
       this.price = price;
       this.stock = stock;
       this.image = image;
       this.ingredients = ingredients;
       this.category = category;
   } 
   public Product( string name, string description, double price, int stock, string image, string ingredients, Category category)
   {
       this.name = name;
       this.description = description;
       this.price = price;
       this.stock = stock;
       this.image = image;
       this.ingredients = ingredients;
       this.category = category;
   }
   
   public Product() { }

   public Product(string name, string description, double price, string image, string ingredients, Category category)
   {
       this.name = name;
       this.description = description;
       this.price = price;
       this.image = image;
       this.ingredients = ingredients;
       this.category = category;
   }

   public Product(int stock)
   {
       this.stock = stock;
   }
}

File: Java/Supplier/src/main/java/domain/Product.java
package domain;

public class Product {

    public int id;
    public String name;
    public String description;
    public double price;
    public Category category;

    public Product(int id, String name, String description, double price, Category category){

        this.id=id;
        this.name=name;
        this.description=description;
        this.price=price;
        this.category=category;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public int getId() {
        return id;
    }

    public Category getCategory() {
        return category;
    }

    public double getPrice() {
        return price;
    }

    public String getDescription() {
        return description;
    }

    public String getName() {
        return name;
    }


}
